httpclient/request_id_round_tripper.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"context"
+	"github.com/acronis/go-appkit/httpserver/middleware"
+	"net/http"
+)
+
+// RequestIDRoundTripper for X-Request-ID header to the request.
+type RequestIDRoundTripper struct {
+	// Delegate is the next RoundTripper in the chain.
+	Delegate http.RoundTripper
+
+	// Opts are the options for the request ID round tripper.
+	Opts RequestIDRoundTripperOpts
+}
+
+// RequestIDRoundTripperOpts for X-Request-ID header to the request options.
+type RequestIDRoundTripperOpts struct {
+	// RequestIDProvider is a function that provides a request ID.
```

Comment:
Please include information in the comment that, by default, `middleware.GetRequestIDFromContext` is used.